src/zarr/storage/common.py

```diff
@@ -209,37 +214,44 @@ async def make_store_path(
     from zarr.storage.remote import RemoteStore  # circular import
 
     used_storage_options = False
-
+    path_normalized = normalize_path(path)
     if isinstance(store_like, StorePath):
         if mode is not None and mode != store_like.store.mode.str:
             _store = store_like.store.with_mode(mode)
             await _store._ensure_open()
-            store_like = StorePath(_store)
-        result = store_like
+            store_like = StorePath(_store, path=store_like.path) / path_normalized
+        result = store_like / path_normalized
```

Comment:
looks like `/ path_normalized` is being applied twice here.

Reply:
good catch. I noticed that this wasn't failing any tests, so in addition to fixing this, I added tests for it in https://github.com/zarr-developers/zarr-python/pull/2384/commits/e6ecf99e67ef8e046214eeec297c014e367e59e0